lib/st/api/lims/samplesheet.pm

```diff
@@ -315,7 +315,7 @@ for my $m ( st::api::lims->driver_method_list_short(__PACKAGE__->meta->get_attri
           if (defined $value && $value eq q[]) {
             $value = undef;
           }
-          if ($m =~ /s$/smx) {
```

Comment:
Is `email_addresses` the only possible "plural" in our samplesheet (de)serialisation?

Reply:
Could we add a code comment  - this is to deal with possible multiple values ?

Alternative would be `($m =~ /s$/smx && $m !~ /lims$/smx)` I suppose.

Not bothered about this alternative (unless you like it) - but think I'd find the comment useful.

Reply:
what went wrong if we left it as it was? i.e. just `($m =~ /s$/smx)`

Reply:
The value of sample_lims were returned as an array when the samplesheet driver was used and as a scalar when teh mlwh driver was used.

Reply:
Some [documentation ](https://github.com/wtsi-npg/npg_tracking/pull/853/commits/c78d854e1e3bf820e7ea8e3e3cd56ecffe9163bf)is added

Reply:
One way of another the regexp has to mention one of the properties explicitly, either emails or sample_lims. I believe allowing for properties with names ending with 's' is a better option.